extensions/machine-learning/src/common/utils.ts

```diff
@@ -65,10 +65,24 @@ export function getPythonInstallationLocation(rootFolder: string) {
 }
 
 export function getPythonExePath(rootFolder: string): string {
-	return path.join(
-		getPythonInstallationLocation(rootFolder),
-		constants.pythonBundleVersion,
+	// The bundle version (0.0.1) is removed from the path for ADS-Python 3.8.8+.
+	// Only ADS-Python 3.6.6 contains the bundle version in the path.
+	let oldPythonPath = path.join(
+		rootFolder,
+		'0.0.1',
+		process.platform === constants.winPlatform ? 'python.exe' : 'bin/python3');
+	let newPythonPath = path.join(
+		rootFolder,
 		process.platform === constants.winPlatform ? 'python.exe' : 'bin/python3');
+
+	// Note: If Python exists in both paths (which can happen if the user chose not to remove Python 3.6 when upgrading),
+	// then we want to default to using the newer Python version.
+	if (!fs.existsSync(newPythonPath) && !fs.existsSync(oldPythonPath) || fs.existsSync(newPythonPath)) {
```

Comment:
Why are you using the sync version of fs.exists?

Reply:
Product code should use promised versions